### PR TITLE
Create abutton UI component for sort button

### LINKED ISSUE #235: 💄 Creating a UI Component for Sort button
We want to have a Sort Button dropdown UI component so that we can have a consistent design. Something like below: 

image

### PR DESCRIPTION
**Pull request by jazellemaira**
This PR creates a UI component for a sort button. Fixes #235 

Needs hooking up @nsunami  🙏🏽 

Screenshot 2024-10-06 at 20 38 19

### CODE REVIEW

**Comment by nsunami** [comment]
@jazellemaira  Yay! Let's implement in a separate PR. Reviewing the component now.

File: web/src/components/SortButton/SortButton.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <Select>
+      <SelectTrigger className="w-[210px]">
+        <SelectValue placeholder="Sorteren" />
+      </SelectTrigger>
+      <SelectContent>
+        <SelectGroup>
+          <SelectLabel>Sorteren op</SelectLabel>
+          <SelectSeparator />
+          <SelectItem value="latest">Laatste</SelectItem>
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.tsx:24]: I was wondering if rephrasing it to make it super clear. Something like, newest first vs oldest first. What do you think? 

```suggestion
          <SelectItem value="newestFirst">Nieuwste Eerst</SelectItem>
```

File: web/src/components/SortButton/SortButton.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          <SelectItem value="oldest">Oudste</SelectItem>
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.tsx:25]:
```suggestion
          <SelectItem value="oldestFirst">Oudste Eerst</SelectItem>
```

File: web/src/components/SortButton/SortButton.tsx (hunk 1 above)
Comment (nsunami) [reply web/src/components/SortButton/SortButton.tsx:24]: "Meest recent" may also be a good idea

File: web/src/components/SortButton/SortButton.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          <SelectItem value="recentlyUpdated">Onlangs bijgewerkt</SelectItem>
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.tsx:26]:
```suggestion
          <SelectItem value="recentlyUpdatedFirst">Laatste update eerst</SelectItem>
```

File: web/src/components/SortButton/SortButton.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          <SelectItem value="leastRecentlyUpdated">
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.tsx:27]:
```suggestion
          <SelectItem value="mostStaleFirst">
```

Is "stale" a bit cheeky?hehe

File: web/src/components/SortButton/SortButton.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            Minst onlangs bijgewerkt
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.tsx:28]:
```suggestion
            Oudste update eerst 
```

File: web/src/components/SortButton/SortButton.test.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { render } from '@redwoodjs/testing/web'
+
+import SortButton from './SortButton'
+
+//   Improve this test with help from the Redwood Testing Doc:
+//    https://redwoodjs.com/docs/testing#testing-components
+
+describe('SortButton', () => {
```
Comment (nsunami) [inline web/src/components/SortButton/SortButton.test.tsx:8]: ✅ Test is passing!

File: web/src/components/ui/select.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+const SelectGroup = SelectPrimitive.Group
+
+const SelectValue = SelectPrimitive.Value
+
+const SelectTrigger = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Trigger>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger>
+>(({ className, children, ...props }, ref) => (
+  <SelectPrimitive.Trigger
```
Comment (nsunami) [inline web/src/components/ui/select.tsx:23]: I got a strange situation where focusing on the button will change the border radius of the button to more rounded (kind of full?). Not critical. But I just wanted to check in to see if you have any ideas on why it's happening. 🤞

image

**Review by nsunami — CHANGES_REQUESTED** [review]
Great work! 🙌 Could you please check my suggestions, and let me know if they make sense?

File: web/src/components/SortButton/SortButton.tsx (hunk 4 above)
Comment (jazellemaira) [reply web/src/components/SortButton/SortButton.tsx:27]: stale does sound like a diss haha

**Review by nsunami — APPROVED** [review]
Looks good to me!

**Pull request merged**